### PR TITLE
docs(developers): add basic DEVELOPER.md and update README

### PR DESCRIPTION
**Pull request by juliemr**

### CODE REVIEW

**Comment by juliemr** [comment]
@sjelin @cnishina @mgiambalvo all suggestions welcome!

File: DEVELOPER.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+## Formatting
+
+Protractor uses clang-format to format the source code. If the source code is not properly formatted,
+the CI will fail and the PR can not be merged.
+
+You can automatically format your code by running:
+
+```
+npm run format
+```
```
Comment (sjelin) [inline DEVELOPER.md]: You can also run linting with `$(npm bin)/gulp lint`

File: DEVELOPER.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+working with selenium-webdriver. github.com/angular/jasminewd
+
+`blocking-proxy` is a separate binary, which handles traffic between a test script and
+webdriver. It can be turned on via a protractor configuration file, and in the future
+all logic to wait for Angular will be handled through the blocking proxy.
+github.com/angular/blocking-proxy
+
+`webdriver-js-extender` extends selenium-webdriver to add Appium commands.
+github.com/angular/webdriver-js-extender
+
```
Comment (sjelin) [inline DEVELOPER.md]: no `webdriver-manager`?

File: DEVELOPER.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+## Continuous Integration
+
+PRs or changes submitted to master will automatically trigger contiuous integration on two
+different services - Travis, and Circle CI.
+
+Travis runs e2e tests via SauceLabs against a variety of browsers. The essential browsers run a
+more complete test suite. We also run a set of smoke tests against a larger set of browsers,
+which is allowed to fail. This is due to flakiness in IE, Safari and older browser versions.
```
Comment (mgiambalvo) [inline DEVELOPER.md]: Maybe call out the specific conf files here - ciFull, ciSmoke, etc. Also what's up with BrowserStack? Is that intended to work? It seems to flake slightly more often than Travis.

File: DEVELOPER.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+Circle CI runs a slightly modified version of `npm test` in a single VM. It installs
+the browsers it needs locally. Circle CI runs unit tests and a set of e2e tests against Chrome.
```
Comment (mgiambalvo) [inline DEVELOPER.md]: Maybe mention why we have two systems? Circle limits you by running instance time, so it's better to keep the Saucelabs tests running on Travis. Also, stdout and stderr logs of all processes as well as a tcpdump of webdriver traffic are all available in the CircleCI artifacts for each build.

File: DEVELOPER.md (hunk 1 above)
Comment (juliemr) [inline DEVELOPER.md]: Done

File: DEVELOPER.md (hunk 2 above)
Comment (juliemr) [inline DEVELOPER.md]: Doh. Added.

File: DEVELOPER.md (hunk 3 above)
Comment (juliemr) [inline DEVELOPER.md]: Done

File: DEVELOPER.md (hunk 4 above)
Comment (juliemr) [inline DEVELOPER.md]: done

**Pull request merged**